### DISCUSSION TITLE
Dark theme for calendar?

### DISCUSSION THREAD

**Discussion by bsktcase** [coedit/gcal-rainbow-surf#4] [archive]
I use Dark Reader, which is tolerable on Google Calendar generally but not nice at all with Rainbow.surf (the blue weekends are especially awful in its dark mode). If Rainbow.surf could offer a dark theme with coordinated events and weekends color palette, it could look amazing and be much easier on our eyes! :)

**Comment by ezdub** [comment]
Hello, welcome to the Rainbow.surf discussion & thank you for the feedback!  

I also use Dark Reader, but am not experiencing anything awful :D 
Weekends are not blue in dark scheme here but they are less black than the rest of the week.

I've just reset my Dark Reader theme to make sure I wasn't doing anything special & it still looks good.
Perhaps you could set Dark Reader to use the default theme on calendar.google.com 
- Let me know if you would like help; I think site-specific settings only appear after clicking Dev Tools > Switch to New Design

If this isn't working I think attaching a screenshot of what you're experiencing is a good place to start.
Ciao! -Z

**Comment by bsktcase** [comment]
Oh no - I've discovered it's intermittent!

Sometimes this (which is painful): ![image](https://user-images.githubusercontent.com/13561511/193163438-010cf4a2-7466-4722-a1f0-873472417c89.png)

But other times and without doing anything at all, this (which is lovely): 
![image](https://user-images.githubusercontent.com/13561511/193163505-ac4d22bb-82c6-4976-8725-f9b3982c6f29.png)

**Reply by ezdub** [reply]
Lovely! Thank you for documenting this. Now I've caught it acting up here. Intermittently.
Steps most likely to reproduce: 

1. Switch to Month view — <kbd>M</kbd>
2. Reload page
3. Switch to Week view — <kbd>W</kbd>

**Quick fix**: press the key for the view you are already in — <kbd>M</kbd> / <kbd>W</kbd> / <kbd>D</kbd> / etc — it does a quick repaint & color is corrected. 

Code fix:
I suspect a MutationObserver in Dark Reader extension is missing what's changed when our MutationObserver fires. I don't think there's much recourse there but we can target our CSS for Dark Reader in dark mode with _html[data-darkreader-scheme="dark"] {}_
Happy to have a PR if anyone can gets to a solution.

**Comment by ezdub** [comment]
This should all be fixed without anything intermittent after PR #17 which is version 2.4.1 published in the Chrome Web Store.
Please report if anything else glitchy happens
